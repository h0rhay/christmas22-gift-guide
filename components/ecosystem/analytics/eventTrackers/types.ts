import { ClickEventActions } from "./eventsConstants";
import { EventTrackerType } from "./eventTracker";

export type GoogleAnalyticsType = {
  ga_event_name: string;
  ga_eventcategory?: string;
  ga_eventaction?: ClickEventActions;
  ga_eventlabel?: string;
};

export type CommonEventsTypes = {
  page_name: string;
  page_type?: string;
}

export type ViewEventTrackerType = {
  // specific for view event
  pageload_event?: string;
  page_load_time?: string; // not used anymore?
  page_subtype?: string;
  page_breadcrumb?: string
  page_category_name?: string
} & CommonEventsTypes
  & GoogleAnalyticsType
  & EventTrackerType

export type LinkEventTrackerType =
  CommonEventsTypes
  & GoogleAnalyticsType
  & EventTrackerType
  & Omit<EventTrackerType, 'event_name' | 'ga_event_name' | 'ga_eventcategory' | 'ga_eventaction'>;

export type DynamicViewEventTrackerType = {
  page_name: string;
}

export type DynamicLinkEventTrackerType = {
  ga_eventaction?: ClickEventActions;
  page_name: string;
  ga_eventlabel?: string;
}
