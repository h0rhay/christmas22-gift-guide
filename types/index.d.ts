import React, { SVGProps } from "react";

declare global {
  interface Window {
    utag: {
      link: Function;
      view: Function;
    };
  }
}

export type IStock = {
  "Stock Quantity Available to Purchase": string;
};

export type IProductDescription = {
  id: string;
  pdpLink?: string;
  imageLink?: string;
  brand: string | boolean;
  productName?: string;
};

export type IPrice = {
  price: string | boolean;
};

export type InitialProps = {
  json: Promise<any>;
  statusCode: boolean;
};
export type CarouselPositionTypes = keyof CarouselsProductsTypes;

type UtagType = {
  link: Function;
  view: Function;
};
export type CarouselsProductsTypes = {
  top?: IProductDetails[];
  bottom?: IProductDetails[];
};

export type IProductDetails = {
  pdpLink?: string;
  imageLink?: string;
  brand: string | boolean;
  productName?: string;
  price: string | boolean;
  altText?: string;
};

export type RandomImagePathType = {
  src: string;
  alt: string;
};

export type ILandingPageSchema = {
  id?: number;
  title?: string;
  desc: string;
  pullQuote: string;
  landingPageImages: RandomImagePathType[];
};

export type VideoType = {
  imgId: string;
  landscapeVideoId: string;
  portraitVideoId: string;
  description?: string;
};

export type IContent = {
  image: {
    url: string;
    type: string;
    credit?: string;
  };
  video: VideoType;
  sticker: StickerType;
  baubleType: BaubleType;
  title: string;
  subtitle: string;
  step?: boolean;
  description: string;
  button: {
    url: string;
    text: string;
  };
  className: string;
};

export type ContentComponentType =
  | "iconTitle"
  | "row-reverse"
  | "row"
  | "stack"
  | "sponsor"
  | "sponsor-reverse";

export type IContentComponent = {
  type: ContentComponentType;
  content: IContent[];
};

export type ICategorySchema = {
  id?: number;
  slug?: string;
  content?: string;
  title?: string;
  subtitle?: string;
  type?: string;
  theme?: string;
  desc?: string;
  tasteMaker?: string;
  hoverImage?: string;
  image?: {
    src: string;
    alt: string;
    credit?: string;
  };
  heroCta?: {
    link: string;
    text: string;
  };
  topHeading?: {
    content: ITopHeading[];
  };
  carousel?: CarouselDataType;
  contentComponent?: IContentComponent[];
  pullQuote?: string;
  landingPageImages?: RandomImagePathType[];
  video?: "video";
};

export type ITopHeading = {
  baubleType: BaubleType;
  title: string;
  description: string;
};

export type CarouselDataType = {
  top?: string[];
  bottom?: string[];
};

export type StickerType = {
  id: number;
  type: StickersContentTypes;
  theme: string;
  title?: string;
  baubleType: BaubleType;
  content?: Array<string> | string;
  button?: ButtonPropType;
};

export type ButtonPropType = {
  url: string;
  text: string;
};
export type BaubleType =
  | "candle"
  | "milk-cookie"
  | "misletoe"
  | "pudding"
  | "stocking"
  | "tree"
  | "wreath";

export type StickersContentTypes = "recipe" | "comment";

export type IKeyOfCategory = keyof ICategorySchema;

export type FramePathType = {
  className?: string;
};

export type FrameOptionsType = "mirror" | "painting" | "cloud";

export type FrameType = {
  type: FrameOptionsType;
};

export type PrimitiveGreyColorType = {
  main: string;
  mid: string;
  light: string;
};

export type PrimitiveColorsType = {
  red: string;
  bubblegum: string;
  olive: string;
  yolk: string;
  sfYellow: string;
  milk: string;
  milkTransparent: string;
  whiteTransparent: string;
  black: string;
  white: string;
  grey: PrimitiveGreyColorType;
};

export type ColorsGroupType = {
  [colorGroupName: string]: ColorGroupType;
};

export type TypographyType = {
  fontFamily: FontFamilyType;
  fontWeight: FontWeightType;
  fontSize: FontSizeType;
  headingFontSize: CommonSizesType;
  lineHeight: CommonSizesType;
  letterSpacing: CommonSizesType;
  fontStyle: FontStyleType;
  space: CommonSizesType;
};

export type FontFamilyType = {
  formula: string;
  apoc: string;
  universalSansDisplay: string;
  avalonDemi: string;
  default: string;
};

export type FontWeightType = {
  light: number;
  regular: number;
  medium: number;
  bold: number;
};

export type FontSizeType = {
  small: string;
  large: string;
  xlarge: string;
  medium: string;
};

export type FontStyleType = {
  normal: string;
  italic: string;
};

/**Icons types from node_modules */
export interface IconBaseProps extends React.SVGAttributes<SVGElement> {
  children?: React.ReactNode;
  size?: string | number;
  color?: string;
  title?: string;
}
export declare type IconType = (props: IconBaseProps) => JSX.Element;
export declare function IconBase(
  props: IconBaseProps & {
    attr?: {};
  }
): JSX.Element;
