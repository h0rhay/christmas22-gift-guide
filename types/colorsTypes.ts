export type ColorGroupType = {
  surface: string;
  interface: string;
};

export type PrimitiveGreyColorType = {
  main: string;
  mid: string;
  light: string;
};

export type PrimitiveColorsType = {
  red: string;
  bubblegum: string;
  olive: string;
  yolk: string;
  sfYellow: string;
  milk: string;
  grey: PrimitiveGreyColorType;
};
