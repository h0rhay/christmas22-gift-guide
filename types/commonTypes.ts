export type CommonSizesType = {
  tinyGap?: "0.5rem";
  extraSmallGap?: string;
  smallGap?: string;
  mediumGap?: string;
  largeGap?: string;
  small: string;
  medium: string;
  large: string;
  extraSmall?: string;
  xlarge?: string;
};

export type LineHeight = CommonSizesType & {
  heading: string;
};

const commonSizesEntryItems = [
  "small",
  "medium",
  "large",
  "xlarge",
  "extraSmall",
] as const;
export type CommonSizesEntriesType = typeof commonSizesEntryItems[number];
