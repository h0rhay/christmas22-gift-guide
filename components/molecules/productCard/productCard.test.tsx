import { render } from "@testing-library/react";
import { describe, expect, it } from "@jest/globals";
import { ThemeProvider } from "styled-components";
import Theme from "styles/theme";
import ProductCard from "./";
import { IProductDetails } from "types";

const renderComponent = () => {
  const product = {
    brand: "Test Brand",
    productName: "Test Product Name",
    price: "Â£300",
    imageLink:
      "https://images.selfridges.com/is/image/selfridges/5120-10004-4438440879_M?PDP_M_ZOOM",
    pdpLink:
      "https://www.selfridges.com/GB/en/cat/burberry-furley-checked-sliders_5120-10004-4438440879",
  } as IProductDetails;

  return render(
    <ThemeProvider theme={Theme}>
      <ProductCard product={product} />
    </ThemeProvider>
  );
};

describe("ProductCard", () => {
  it("should match snapshot", () => {
    const { container } = renderComponent();
    expect(container).toMatchSnapshot();
  });
});
