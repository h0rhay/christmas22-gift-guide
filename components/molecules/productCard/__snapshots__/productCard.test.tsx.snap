// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCard should match snapshot 1`] = `
<div>
  <div
    class="productCard__StyledProductCard-sc-phbwqf-0 hNMhFP"
  >
    <div
      class="imageComponent__StyledImage-sc-y6629v-0 BNymA image-component__container"
      type="product"
    >
      <div
        style="display: block; overflow: hidden; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; margin: 0px;"
      >
        <img
          alt="Test Brand Test Product Name"
          class=""
          data-nimg="fill"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; filter: blur(20px); background-size: cover; background-image: url(https://images.selfridges.com/is/image/selfridges/5120-10004-4438440879_M?PDP_M_ZOOM); background-position: 0% 0%;"
        />
        <noscript />
      </div>
    </div>
    <div
      class="productCard__ProductDetails-sc-phbwqf-1 cPSSYZ"
    >
      <h4
        class="productCard__BrandName-sc-phbwqf-2 cuXzdU"
      >
        Test Brand
      </h4>
      <p>
        Test Product Name
      </p>
      <p
        class="product-card-price"
      >
        £
        £300
      </p>
    </div>
    <div
      class="productCard__ButtonContainer-sc-phbwqf-3 eqmKwk"
    >
      <button
        class="button__ButtonStyles-sc-lri851-0 bPsHIA"
        data-test-id="reverse"
        type="button"
      >
        <a
          href="https://www.selfridges.com/GB/en/cat/burberry-furley-checked-sliders_5120-10004-4438440879"
          target="_blank"
        >
          Shop Now
        </a>
      </button>
    </div>
  </div>
</div>
`;
